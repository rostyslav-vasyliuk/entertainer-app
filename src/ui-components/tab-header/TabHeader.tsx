import React from 'react';
import { Text, StyleSheet, View } from 'react-native';
import { Icon } from 'native-base';
import { TEXT_COLOR } from '../../constants/color-constants';

const TabHeaderLabel = (props) => {
  return (
    <View style={styles.wrapper}>
      <Text style={styles.text}>
        {props.text}
      </Text>
      <Icon name={props.icon} style={styles.icon} />
    </View>
  )
}

export default TabHeaderLabel;

const styles = StyleSheet.create({
  text: {
    color: TEXT_COLOR,
    fontSize: 16,
    letterSpacing: 1.1,
    fontWeight: '600',
    overflow: 'visible',
    paddingRight: 6
  },
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 80,
    width:200
  },
  icon: {
    fontSize: 20,
    paddingTop: 2,
    color: TEXT_COLOR

  }
})
