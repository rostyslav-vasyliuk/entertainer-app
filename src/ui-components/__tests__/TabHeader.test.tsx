import React from 'react';
import TabHeader from '../tab-header/TabHeader';
import renderer from 'react-test-renderer';

describe('<TabHeader />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<TabHeader />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it('has 1 child', () => {
    const tree = renderer.create(<TabHeader />).toJSON();
    expect(tree.children.length).toBeGreaterThanOrEqual(0);
  });

  it('has 1 child', () => {
    const tree = renderer.create(<TabHeader />).toJSON();
    expect(tree.children.length).toBeGreaterThanOrEqual(1);
  });

  it('rendered', () => {
    const tree = renderer.create(<TabHeader />).toJSON();
    expect(tree.children).toBeTruthy();
  });

  it('to match snapshot', () => {
    const tree = renderer.create(<TabHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
