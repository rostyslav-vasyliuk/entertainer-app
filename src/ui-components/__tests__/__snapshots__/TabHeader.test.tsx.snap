// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabHeader /> to match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "600",
        "letterSpacing": 1.1,
        "overflow": "visible",
        "paddingRight": 6,
      }
    }
  />
  <Text />
</View>
`;
